### ISSUE TITLE
Tensor Pointcloud to PLY writes non-standard types 

### ISSUE THREAD

**Issue by TheExDeus** [isl-org/Open3D#4548]
The header generated when writing using `o3d.t.io.write_point_cloud` uses data types like "float32" and "uint8". These don't seem to be standard types. There are tools which can open that, but as far as I understand the only valid types are:
```
name        type        number of bytes
---------------------------------------
char       character                 1
uchar      unsigned character        1
short      short integer             2
ushort     unsigned short integer    2
int        integer                   4
uint       unsigned integer          4
float      single-precision float    4
double     double-precision float    8
```
That comes from http://paulbourke.net/dataformats/ply/, https://se.mathworks.com/help/vision/ug/the-ply-format.html and many more. I haven't seen a source mentioning specific sized types. Maybe it's legal as there isn't really a strict "standard", but it's still unusual. No other tools that I have checked save like that, like PCL and Meshlab.

Also this is a different behavior from `o3d.io.write_point_cloud` (non-Tensor) variant as that writes standard names.

It makes sense that maybe a custom type not in these specified ones are written like this. Or if the user could name the type as required.

**To Reproduce**
Use `o3d.t.io.write_point_cloud()` to write a PLY file in either ASCII or Binary. Header will have sized types.

**Expected behavior**
All types should be as standard.

Example header when saving with `o3d.io.write_point_cloud` (non-Tensor):
```
ply
format binary_little_endian 1.0
comment Created by Open3D
element vertex 168230
property double x
property double y
property double z
property uchar red
property uchar green
property uchar blue
end_header
```

Header when saving with `o3d.t.io.write_point_cloud` (Tensor):
```
ply
format binary_little_endian 1.0
comment Created by Open3D
element vertex 168230
property float32 x
property float32 y
property float32 z
property uint8 red
property uint8 green
property uint8 blue
end_header
```

**Comment by reyanshsolis** [comment]
[rply.h Line 37: ply data type](https://github.com/isl-org/Open3D/blob/c91e0a729e86b7aeda22c428c681709d2714fbf1/3rdparty/rply/rply/rply.h#L37)
```
    PLY_INT8, PLY_UINT8, PLY_INT16, PLY_UINT16, 
    PLY_INT32, PLY_UIN32, PLY_FLOAT32, PLY_FLOAT64,
    PLY_CHAR, PLY_UCHAR, PLY_SHORT, PLY_USHORT,
    PLY_INT, PLY_UINT, PLY_FLOAT, PLY_DOUBLE,
    PLY_LIST    /* has to be the last in enum */
``` 
These are the types that are supported by `RPLY`, therefore these must be standard.
So, `double` and `float64`, `float` and `flaot32`, `uint8` and `uchar` are equivalent.  

I may update `Float32` to be saved as `float`, and `Double` to be saved as `double`. But, even `float32` and `float64` are standard formats. You should not have any problem opening the files written by Open3D in Meshlab or PCL (if there is please report the same, I can take a look). 

## Other changes.
### OLD
Legacy / Eigen based API supported only `double` format. 
Therefore everything was converted to double (and colours were converted to `uchar` while writing). So, if you read a `.ply` file in legacy and write it back to `.ply` in legacy, the positions and normals will be saved as `double` and colors will be saved as `uchar`, irrespective of the original format. Even if you read from a `.ply` file with  `float` positions, the type will be `double`. 

### New (Tensor)
Now, more formats and attributes are supported. 
-  if the input file has float, it will be read as float, and saved as a float. 
- Similarly for colours and other attributes.
- Now even custom attributes such as `intensity`, `curvature`, or any random attribute can also be saved, which was previously not possible.

**Comment by TheExDeus** [comment]
Yes, I noticed that legacy API read floats as doubles and then saved doubles. That is how I got to this question, because when I switched to Tensor API I saw the difference and I couldn't find other tools which save like that. I don't have many tools to test for loading compatibility, but the ones I tried worked. So I guess it could be "standard". I guess this is the issue with PLY not really having a written enforced standard. Most just reference a blog. :D

Thanks for the info and changes!
